src/web_app/web/reminder_view.py

```diff
@@ -0,0 +1,136 @@
+from flask import Blueprint, render_template, request, Response
+
+from src import Logger, log_path
+from src.main import client
+from src.main.objects.reminder_manager import ReminderManager
+from src.models.base import DataBase
+
+app_reminder = Blueprint('app_reminder', __name__, template_folder='templates')
+logger = Logger(log_path)
+reminder_manager = ReminderManager(client, DataBase(), logger)  # used for all interactions with database
+
+@app_reminder.route('/ui/reminder', methods=['GET'])
+def get_reminders():
+	"""
+	default mapping with all reminders
+	"""
+	reminders = reminder_manager.get_reminders()
+	courses, authors = reminder_manager.get_filter_options()
+	return render_template('reminder.html', courses=courses, authors=authors, reminders=reminders), 200
+
+
+@app_reminder.route('/ui/reminder/filter', methods=['POST'])
+def filter_reminders():
+	"""
+	mapping for returning only filtered reminders
+	params: name : name of the course
+			author : first and last name of the author
+			from : all dates after it
+			to : all dates after it
+			posted : true or false, reminder posted in workspace or not
+	returns: dict that has form for loading bootstrap table
+			result : number of reminders
+			rows : all data for reminders, must be JSON serializable
+	"""
+	filters = dict()
+	if len(request.args) != 0:
+		for elem in ['name', 'author', 'from', 'to', 'posted']:
+			filters[elem] = request.args.get(elem)
+
+		# getting rid of not selected filters
+		filters.pop('name') if filters['name'] == "Nothing selected" else None
+		filters.pop('author') if filters['author'] == "Nothing selected" else None
+		filters.pop('from') if not filters['from'] else None    # default is empty string
+		filters.pop('to') if not filters['to'] else None
+
+		reminders = reminder_manager.get_reminders(**filters)
+
+	# canceling filters and returning all reminders
+	else:
+		reminders = reminder_manager.get_reminders()
+
+	reminders_json = list()
+	# turn a list of reminders into a list of reminder attributes in dict form
+	# end_date and timer are parsed to string because timestamp and interval are not json serializable
+	for reminder in reminders:
+		reminders_json.append({'id': reminder.id,
+		                       'end_date': str(reminder.end_date),
+		                       'timer': str(reminder.timer),
+		                       'text': reminder.text,
+		                       'posted': reminder.posted})
+	return {'result': len(reminders_json), 'rows': reminders_json}
```

Comment:
šta ne možeš tu predat objekt Reminder jer je to isto kao da predaš dict

Reply:
Ne može jer objekt Reminder ima TIMESTAMP i INTERVAL koji nisu JSON serializable